### PR TITLE
Solution for broken SMP DQM codes for validations 

### PR DESCRIPTION
**Pull request by sasharma6**
" Poblem : There were missing EwkElecDQM & EwkMuonDQM modules in the physics directory of samples ZEE, ZMM, WE WM. 
Which leads to no validation of CMSSW releases for 81X and few 80X.
Solution implemented : As suggested by the group people we have removed the trigger part from both EwkElec and EwkMuon DQM codes and using CMSSW 810pre9 we obtained the plots.
Things to do: To approve the code and check whether it is properly merged"

### CODE REVIEW

**Comment by dmitrijus** [comment]
Please set the proper title and the detailed description.
For example, such as in this pull request: https://github.com/cms-sw/cmssw/pull/15830

**Comment by sasharma6** [comment]
Hi Dmitrijus,

I have added the information for PR#16322. Kindly have a look.

Thanks and regards
Sandeep

Get Outlook for Android<https://aka.ms/ghei36>



On Fri, Oct 28, 2016 at 7:18 PM +0530, "Dmitrijus" <notifications@github.com<mailto:notifications@github.com>> wrote:


Please set the proper title and the detailed description.
For example, such as in this pull request: #15830<https://github.com/cms-sw/cmssw/pull/15830>

File: DQM/Physics/src/EwkElecDQM.cc [hunk 1]
```diff
@@ -1,1039 +1,955 @@
...
+    snprintf(chtitle, 255, "Eta of Jet with highest E_{T} (%s)", jetTag_.label().data());
+    jet_eta_before_ = ibooker.book1D("JETETA1_BEFORECUTS", chtitle, 20, -5, 5);
+    jet_eta_after_ = ibooker.book1D("JETETA1_AFTERCUTS", chtitle, 20, -5, 5);
+}
+
+void EwkElecDQM::endRun(const Run & r, const EventSetup &)
+{
+    double all = nall;
+    double esel = nsel / all;
+    LogVerbatim("") << "\n>>>>>> SELECTION SUMMARY BEGIN >>>>>>>>>>>>>>>";
```
Comment (dmitrijus) [inline DQM/Physics/src/EwkElecDQM.cc]: Why does this use LogVerbatim? Why not LogWarning?

File: DQM/Physics/src/EwkElecDQM.cc [hunk 2]
```diff
@@ -1,1039 +1,955 @@
...
-      jet_et = jet.et();  // current highest jet gets et from the new highest
-      jet_eta = jet.eta();
-    } else if (jet_current_et > jet2_et) {
-      jet2_et = jet.et();
+    unsigned int electronCollectionSize = electronCollection->size();
+
+    // Beam spot
+    Handle < reco::BeamSpot > beamSpotHandle;
+    if (!iEvent.getByToken(beamSpotTag_, beamSpotHandle)) {
+        std::cout << ">>> No beam spot found !!!";
```
Comment (dmitrijus) [inline DQM/Physics/src/EwkElecDQM.cc]: No std::cout or std::cerr!

File: DQM/Physics/src/EwkElecDQM.cc [hunk 3]
```diff
@@ -1,1039 +1,955 @@
...
+        if (sieie < sieieCutEndcap_ && isEndcap)
+            electron_sel[2] = true;
+        if (detain < detainCutBarrel_ && isBarrel)
+            electron_sel[3] = true;
+        if (detain < detainCutEndcap_ && isEndcap)
+            electron_sel[3] = true;
+        int out2 = 0;
+        if (out2 == 1) {
+            if (isBarrel) {
+                std::cout << "\t... detain value " << detain << " (barrel), pass? " << electron_sel[3];
```
Comment (dmitrijus) [inline DQM/Physics/src/EwkElecDQM.cc]: std::cout used again!

File: DQM/Physics/src/EwkMuDQM.cc [hunk 4]
```diff
@@ -1,878 +1,967 @@
...
+{
+}
+
+// ------------ method called for each event  ------------
+void EwkMuDQM::analyze(const edm::Event & iEvent, const edm::EventSetup & iSetup)
+{
+    using namespace edm;
+    Handle < View < Muon > >muonCollection;
+    if (!iEvent.getByToken(muonTag_, muonCollection)) {
+        std::cout << ">>> Muon collection does not exist !!!" << std::endl;
```
Comment (dmitrijus) [inline DQM/Physics/src/EwkMuDQM.cc]: !

File: DQM/Physics/src/EwkMuDQM.cc [hunk 5]
```diff
@@ -1,878 +1,967 @@
...
-    if (minDistance < 0.3)
-      continue;  // 0.3 is the isolation cone around the muon
-    if (jet.et() > max_pt) {
-      LEADJET = i;
-      max_pt = jet.et();
+
+    unsigned int muonCollectionSize = muonCollection->size();
+    Handle < reco::BeamSpot > beamSpotHandle;
+    if (!iEvent.getByToken(beamSpotTag_, beamSpotHandle)) {
+        std::cout << ">>> No beam spot found !!!" << std::endl;
```
Comment (dmitrijus) [inline DQM/Physics/src/EwkMuDQM.cc]: !

**Review by dmitrijus — CHANGES_REQUESTED** [review]
Looks much better than previous pull requests.
One major issue is that thread-unsafe std::cout is used. It should be converted into LogWarning (where appropriate). 

Please make a PR to 90x with std::cout removed.
Once it is *merged*, you will be able to submit backports to other branches. However, it is much easier to keep 1 PR for all the discussions.

In general, you should edit the PR instead of making a new for every single change (there is an 'edit' button).

**Comment by dmitrijus** [comment]
Please close this PR. And the other one in 81x.
Your main target is the upstream branch - 90x - https://github.com/cms-sw/cmssw/pull/16630
The PR there has a merge conflict, I cannot start the tests. Fix it and I will start the tests.

**Pull request closed without merging**